src/com/company/asset/Asset.java

```diff
@@ -7,33 +7,48 @@
     private String invoiceDetails;
     private Brand brand;
     private int assetNumber;
-    private int IMEI;
-    private int store=1;
-    private int scrap=0;
+    private TypeOf type;
+    private long IMEI;
+    private boolean assignedTo=false;
+    private boolean isScrapped=false;
 
 
     public Asset() {}
 
-    public Asset(String assetName, int assetNumber, int make, int amount, String invoiceDetails, Brand brand,int imei) {
+    public Asset(String assetName, int assetNumber, int make, int amount, String invoiceDetails, Brand brand,TypeOf type,long imei,boolean assignedTo,boolean isScrapped) {
```

Comment:
No need to acted assignedTo and isScrapped in argument, directly set it to false as done in Line 26-27.
As new asset cannot be scrapped or assigned.